openedx_tagging/core/tagging/system_defined_taxonomies/object_tags.py

```diff
@@ -0,0 +1,210 @@
+    for the same Taxonomy, ex:
+    
+    - LanguageCourseObjectTag
+    - LanguageBlockObjectTag
+
+    On the example, there are ObjectTags for the same Language
+    taxonomy but with different objects.
+
+    Using this approach makes the connection between the ObjectTag
+    and system defined taxonomy as hardcoded and can't be changed.
+    """
+
+    system_defined_taxonomy_id = None
+
+    def _check_system_taxonomy(self, taxonomy: Taxonomy = None):
+        """
+        Validates if the taxonomy is system-defined and match
+        with the name stored in the object tag
+        """
+        return (
+            bool(taxonomy) and
+            taxonomy.system_defined and
+            taxonomy.id == self.system_defined_taxonomy_id
+        )
+
+
+class OpenSystemObjectTag(OpenObjectTag, SystemDefinedObjectTagMixin):
+    """
+    Free-text object tag used on system-defined taxonomies
+    """
+
+    class Meta:
+        proxy = True
+
+    def _check_taxonomy(self):
+        return (
+            super()._check_taxonomy() and
+            self._check_system_taxonomy(self.taxonomy)
+        )
+
+    
+class ClosedSystemObjectTag(ClosedObjectTag, SystemDefinedObjectTagMixin):
+    """
+    Object tags linked to a closed system-taxonomy
+    """
+
+    class Meta:
+        proxy = True
+
+    def _check_taxonomy(self):
+        return (
+            super()._check_taxonomy() and
+            self._check_system_taxonomy(self.taxonomy)
+        )
+
+
+class ModelObjectTag(OpenSystemObjectTag):
+    """
+    Object tags used with tags that relate to the id of a model
+    
+    This object tag class is not registered as it needs to have an associated model
+    """
+
+    class Meta:
+        proxy = True
+
+    tag_class_model = None
+
+
+    def _check_taxonomy(self):
+        """
+        Validates if has an associated Django model that has an Id
+        """
+        if not super()._check_taxonomy():
+            return False
+    
+        if not self.tag_class_model:
+            return False
+
+        # Verify that is a Django model
+        if not issubclass(self.tag_class_model, models.Model):
+            return False
+
+        # Verify that the model has 'id' field
+        try:
+            self.tag_class_model._meta.get_field('id')
+        except FieldDoesNotExist:
+            return False
+
+        return True
+    
+    def _check_instance(self):
+        """
+        Validates if the instance exists
+        """
+        try:
+            intance_id = int(self.value)
+        except ValueError:
+            return False
+        return self.tag_class_model.objects.filter(id=intance_id).exists()
+
+    def _check_tag(self):
+        """
+        Validates if the instance exists
+        """
+        if not super()._check_tag():
+            return False
+        
+        # Validates if the instance exists
+        if not self._check_instance():
+            return False
+        
+        return True
```

Comment:
I like the shorthand implementation you used above better (I should [do that too](https://github.com/openedx/edx-platform/pull/32661#pullrequestreview-1527743164)):
```suggestion
        return (
            super()._check_tag() and
            self._check_instance()
        )
```

Reply:
Updated in all checks